From 7593fbd0c578acf825358755b3e571c8804c1ec8 Mon Sep 17 00:00:00 2001
From: 24bit192kHz <24bit192khz@proton.me>
Date: Wed, 30 Sep 2026 18:40:27 +0300
Subject: [PATCH 12/12] mb/lenovo/sklkbl_thinkpad/t480: Follow the vendor
 platform setup

As on the vendor firmware: the radios are never hard-blocked (the EC has
no wireless switch), no tablet mode switch, the lid wakes from S4, the
GMM device 00:08.0 is enabled, and all PCH and system agent devices
carry the subsystem IDs 17aa:225d.

TEST=T480: Bluetooth initialises and loads its firmware, LID and GLAN
wake from S4, lspci shows 00:08.0 and 17aa:225d on the PCH devices; cold
boot, suspend/resume and hibernate/resume with the dGPU pass.

Change-Id: I8ac9653faec9e77764e27dc3da4593eec458ad37
Signed-off-by: 24bit192kHz <24bit192khz@proton.me>
---
 src/mainboard/lenovo/sklkbl_thinkpad/Kconfig              | 8 ++++++++
 src/mainboard/lenovo/sklkbl_thinkpad/acpi/ec.asl          | 1 +
 .../lenovo/sklkbl_thinkpad/variants/t480/overridetree.cb  | 1 +
 3 files changed, 10 insertions(+)

diff --git a/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig b/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig
index a544def293..e5dbfdedb3 100644
--- a/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig
+++ b/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig
@@ -14,6 +14,8 @@ config BOARD_LENOVO_SKLKBL_THINKPAD_COMMON
 	select H8_HAS_FNLOCK_LED
 	select H8_HAS_PRIMARY_FN_KEYS
 	select H8_SUPPORT_BT_ON_WIFI
+	select H8_NO_WIRELESS_SWITCH
+	select H8_NO_TABLET_MODE_SWITCH
 	select HAVE_ACPI_RESUME
 	select HAVE_ACPI_TABLES
 	select INTEL_GMA_HAVE_VBT
@@ -135,3 +137,9 @@ config VARIANT_HAS_DGPU
 	bool
 
 endif
+
+config SUBSYSTEM_VENDOR_ID
+	default 0x17aa if BOARD_LENOVO_T480
+
+config SUBSYSTEM_DEVICE_ID
+	default 0x225d if BOARD_LENOVO_T480
diff --git a/src/mainboard/lenovo/sklkbl_thinkpad/acpi/ec.asl b/src/mainboard/lenovo/sklkbl_thinkpad/acpi/ec.asl
index 22be737e73..f6d6b7492e 100644
--- a/src/mainboard/lenovo/sklkbl_thinkpad/acpi/ec.asl
+++ b/src/mainboard/lenovo/sklkbl_thinkpad/acpi/ec.asl
@@ -4,6 +4,7 @@
 #define BRIGHTNESS_DOWN \_SB.PCI0.GFX0.DECB
 #define EC_LENOVO_H8_EXTENDED_FN_LAYOUT 1
 #define THINKPAD_EC_GPE 22
+#define H8_LID_WAKE_STATE 4	/* as the vendor firmware */
 
 Name(\TCRT, 100)
 Name(\TPSV, 90)
diff --git a/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480/overridetree.cb b/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480/overridetree.cb
index 17fec4cd26..412bba3268 100644
--- a/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480/overridetree.cb
+++ b/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480/overridetree.cb
@@ -2,6 +2,7 @@
 
 chip soc/intel/skylake
 	device domain 0 on
+		device ref gmm on end	# Gaussian Mixture Model accelerator, as the vendor firmware
 		device ref south_xhci on
 			register "usb2_ports" = "{
 				[0] = USB2_PORT_MID(OC1),		// USB-A
-- 
2.55.0

